PythonVisualizations/PriorityQueue.py

```diff
@@ -0,0 +1,438 @@
+        self.canvas.delete(self.list[toIndex].display_shape)
+
+        # update value and display value in "to" position in the list
+        self.list[toIndex].display_val = newCellVal
+        self.list[toIndex].val = self.list[fromIndex].val
+        self.list[toIndex].display_shape = newCell
+        self.list[toIndex].color = self.list[fromIndex].color
+        callEnviron ^= set([newCell, newCellVal])
+        
+        # delete the original "from" display value and the new display shape
+        self.canvas.delete(self.list[fromIndex].display_val)
+        self.canvas.delete(self.list[fromIndex].display_shape)        
+
+        # update the window
+        self.window.update()
+
+    def cellCoords(self, cell_index):  # Get bounding rectangle for array cell
+        return (self.ARRAY_X0 + self.CELL_SIZE * cell_index, self.ARRAY_Y0,  # at index
+                self.ARRAY_X0 + self.CELL_SIZE * (cell_index + 1) - self.CELL_BORDER,
+                self.ARRAY_Y0 + self.CELL_SIZE - self.CELL_BORDER)
+
+    def cellCenter(self, index):  # Center point for array cell at index
+        half_cell = (self.CELL_SIZE - self.CELL_BORDER) // 2
+        return add_vector(self.cellCoords(index), (half_cell, half_cell))
+
+    def createArrayCell(self, index):  # Create a box representing an array cell
+        cell_coords = self.cellCoords(index)
+        half_border = self.CELL_BORDER // 2
+        rect = self.canvas.create_rectangle(
+            *add_vector(cell_coords,
+                        (-half_border, -half_border,
+                         self.CELL_BORDER - half_border, self.CELL_BORDER - half_border)),
+            fill=None, outline=self.CELL_BORDER_COLOR, width=self.CELL_BORDER)
+        self.canvas.lower(rect)
+        return rect
+
+    def createCellValue(self, indexOrCoords, key, color=None):
+        """Create new canvas items to represent a cell value.  A square
+        is created filled with a particular color with an text key centered
+        inside.  The position of the cell can either be an integer index in
+        the Array or the bounding box coordinates of the square.  If color
+        is not supplied, the next color in the palette is used.
+        An event handler is set up to update the VisualizationApp argument
+        with the cell's value if clicked with any button.
+        Returns the tuple, (square, text), of canvas items
+        """
+        # Determine position and color of cell
+        if isinstance(indexOrCoords, int):
+            rectPos = self.cellCoords(indexOrCoords)
+            valPos = self.cellCenter(indexOrCoords)
+        else:
+            rectPos = indexOrCoords
+            valPos = divide_vector(add_vector(rectPos[:2], rectPos[2:]), 2)
+        if color is None:
+            # Take the next color from the palette
+            color = drawable.palette[self.nextColor]
+            self.nextColor = (self.nextColor + 1) % len(drawable.palette)
```

Comment:
It looks like PriorityQueue is already using the `drawable.palette`.  So there isn't a need for the one created on lines 19-20.